libpolys/polys/monomials/ring.h

```diff
@@ -420,8 +420,13 @@ static inline BOOLEAN rIsLPRing(const ring r)
 
 static inline BOOLEAN rIsNCRing(const ring r)
 {
-#ifdef HAVE_PLURAL
-  return rIsLPRing(r) || rIsPluralRing(r);
+  assume(r != NULL);
+#if defined(HAVE_PLURAL) && defined(HAVE_SHIFTBBA)
```

Comment:
Why so many cases? 
return rIsLPRing(r) || rIsPluralRing(r);
should be correct with and without HAVE_PLURAL and/or HAVE_SHIFTBBA

Reply:
The original version would `return FALSE` when compiled with `HAVE_SHIFTBBA` but without `HAVE_PLURAL `.